### ISSUE TITLE
Ability to extend base AWS Classes with convenience functions

### ISSUE THREAD

**Issue by owap** [OpenWhere/scenery#53]
Develop an elegant solution for extending the automatically-generated Resource classes with convenience functions.

One idea would be to create a `convenience_functions` folder and populate it with files named after their corresponding ResourceType, such as `AWS::EC2::Instance.js`  Then, we could extend the base class to check for these functions similar to how we register property maps.

We should endeavor to keep these convenience functions out of the `scenery_generator` project, because these would be used only in Scenery.

**Comment by owap** [comment]
Here are three convenience functions we used to have in the VPC class:

```

VPC.prototype.addSubnet = function (id, description) {
    var sub = this.template.addResource(new Subnet(id, description, this.template));
    sub.vpcId(this.id);
    return sub;
};

VPC.prototype.addRouteTable = function (id, description) {
    var rt = this.template.addResource(new RouteTable(id, description,this.template));
    rt.vpcId(this.id);
    return rt;
};

VPC.prototype.addNetworkAcl = function (id, description) {
    var nacl = this.template.addResource(new NetworkAcl(id, description,this.template));
    nacl.vpcId(this.id);
    return nacl;
};
```

**Comment by owap** [comment]
We should also consider adding all kinds of UserData-related convenience functions for the AWS::EC2::Instance class!  But those are not requirements to close this issue, which is the implementation of extensible classes with convenience functions.

**Comment by owap** [comment]
**Note:** The above-mentioned VPC helper classes no longer work in the current version of Scenery, because resources are not aware of which template they are in (because you could assign the same resource to multiple templates).

**Comment by owap** [comment]
VPC helper functions have been changed to work with the current template implementation. This was implemented in #60
